src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```
@@ -252,6 +254,54 @@ private function validateAndDenormalize($currentClass, $attribute, $data, $forma
+            if (
+                (XmlEncoder::FORMAT === $format || CsvEncoder::FORMAT === $format) &&
+                \in_array($type->getBuiltinType(), [Type::BUILTIN_TYPE_BOOL, Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_FLOAT]) &&
+                \is_string($data)
+            ) {
+                switch ($type->getBuiltinType()) {
+                    case Type::BUILTIN_TYPE_BOOL:
+                        // according to https://www.w3.org/TR/xmlschema-2/#boolean, valid representations are "false", "true", "0" and "1"
+                        if ('false' === $data || '0' === $data) {
+                            $data = false;
+                        } elseif ('true' === $data || '1' === $data) {
+                            $data = true;
+                        }
+                        break;
+                    case Type::BUILTIN_TYPE_INT:
+
+                        if (
+                            true === ctype_digit($data) ||
+                            '-' == $data[0] && true === ctype_digit(substr($data, 1))
+                        ) {
+                            $data = (int) $data;
+                        }
+                        break;
+                    case Type::BUILTIN_TYPE_FLOAT:
+                        if (is_numeric($data)) {
+                            return '0x' === $data[0].$data[1] ? hexdec($data) : (float) $data;
+                        } elseif (
+                            true === ctype_digit($data) ||
+                            '-' == $data[0] && true === ctype_digit(substr($data, 1))
+                        ) {
+                            $data = (int) $data;
+                        } elseif ('NaN' === $data) {
+                            return NAN;
+                        } elseif ('INF') {
+                            return INF;
+                        } elseif ('-INF') {
+                            return -INF;
+                        }
+
+                        break;
```

so, we silently ignore other values, is this correct? (same for the "int" case btw)

---

better to throw Exception